cogs/study_groups.py

```diff
@@ -134,6 +134,47 @@ async def _group(self, ctx: Context):
         member: Union[Member, User] = ctx.author
         logger.info(f'User="{member.name}#{member.discriminator}({member.id})", Command="{ctx.message.content}"')
 
+    @_group.command(pass_context=True,
+                    name="create",
+                    brief="Creates a Study group.",
+                    help="The name must contain the tag and the semester number.")
+    async def group_create(self, ctx: Context, name: str, color, semester: int = 1):
+        """
+        Adds a new role-channel pair as a group.
+
+        Args:
+            ctx: The command context provided by the discord.py wrapper.
+
+            name: The name of the role and the chat.
+
+            color: color of group.
+
+            semester: number of semester
+        """
+        color_db = PrimitiveMongoData(CollectionEnum.GROUP_COLOR)
+        active_db = PrimitiveMongoData(CollectionEnum.GROUP_ACTIVE)
+        guild: Guild = ctx.guild
+
+        if not await color_db.find_one({name: {"$exists": True}}):
+            await color_db.insert_one({name: color})
+        if not active_db.find_one({name: {"$exists": True}}):
+            await active_db.insert_one({name: True})
+
+        for i in range(1, semester + 1):
+            created = await StudySubjectUtil.get_server_objects(
+                ConfigurationNameEnum.GROUP_CATEGORY,
+                guild,
+                f"{name}{i}",
+                ConfigurationNameEnum.STUDY_SEPARATOR_ROLE,
+                self.db,
+                color=Color(color)
```

Comment:
check if color is an int, if it is a str, try to convert hex to decimal

Reply:
To be more specific, do the check before inserting into database

Reply:
Not necessary type conversion already done by function definition and framework magic.

Reply:
no, not the case, see screenshots:

![image](https://github.com/marcelropos/HM-DiscordBot/assets/44324946/79820444-0600-40a1-b118-036f6192c442)

![image](https://github.com/marcelropos/HM-DiscordBot/assets/44324946/ad5f457a-9659-4f73-aa34-77bde12594b0)

Reply:
You're right. Fixed